From cdb8355a298a084e512704c75dcdcef7bc5f65c6 Mon Sep 17 00:00:00 2001
From: Angie Wang <Angie.Wang@windriver.com>
Date: Tue, 16 Jan 2018 19:20:04 -0500
Subject: [PATCH 77/91] US108388: Enable more polling meters

- Enable more useful polling meters
- Add the missing cpu_delta sink in controller.yaml
---
 etc/ceilometer/controller.yaml | 10 ++++++
 etc/ceilometer/polling.yaml    | 75 ++++++++++++++++++++++++++++++++----------
 2 files changed, 68 insertions(+), 17 deletions(-)

diff --git a/etc/ceilometer/controller.yaml b/etc/ceilometer/controller.yaml
index 4dc37ca..3ba5c76 100644
--- a/etc/ceilometer/controller.yaml
+++ b/etc/ceilometer/controller.yaml
@@ -10,6 +10,7 @@ sources:
           - "cpu"
       sinks:
           - cpu_sink
+          - cpu_delta_sink
           - vcpu_sink
           - csv_sink
     - name: disk_source
@@ -51,6 +52,15 @@ sinks:
                     scale: "100.0 / (10**9 * (resource_metadata.cpu_number or 1))"
       publishers:
           - notifier://
+    - name: cpu_delta_sink
+      transformers:
+          - name: "delta"
+            parameters:
+                target:
+                    name: "cpu.delta"
+                growth_only: True
+      publishers:
+          - notifier://
     - name: vcpu_sink
       transformers:
           - name: "rate_of_change"
diff --git a/etc/ceilometer/polling.yaml b/etc/ceilometer/polling.yaml
index acf786e..bb50bd5 100644
--- a/etc/ceilometer/polling.yaml
+++ b/etc/ceilometer/polling.yaml
@@ -1,29 +1,70 @@
 ---
 sources:
-    - name: some_pollsters
+    - name: instance_pollster
       interval: 600
       meters:
         - memory.usage
-        - network.incoming.bytes
-        - network.incoming.packets
-        - network.outgoing.bytes
-        - network.outgoing.packets
+        - memory.resident
         - disk.read.bytes
+        - disk.read.bytes.rate
         - disk.read.requests
+        - disk.read.requests.rate
         - disk.write.bytes
+        - disk.write.bytes.rate
         - disk.write.requests
-        - hardware.cpu.util
-        - hardware.memory.used
-        - hardware.memory.total
-        - hardware.memory.buffer
-        - hardware.memory.cached
-        - hardware.memory.swap.avail
-        - hardware.memory.swap.total
-        - hardware.system_stats.io.outgoing.blocks
-        - hardware.system_stats.io.incoming.blocks
-        - hardware.network.ip.incoming.datagrams
-        - hardware.network.ip.outgoing.datagrams
-    - name: cpu_pollster
+        - disk.write.requests.rate
+        - disk.capacity
+        - disk.allocation
+        - disk.usage
+    - name: instance_cpu_pollster
       interval: 30
       meters:
         - cpu
+    - name: instance_disk_pollster
+      interval: 600
+      meters:
+        - disk.device.read.requests
+        - disk.device.read.requests.rate
+        - disk.device.write.requests
+        - disk.device.write.requests.rate
+        - disk.device.read.bytes
+        - disk.device.read.bytes.rate
+        - disk.device.write.bytes
+        - disk.device.write.bytes.rate
+        - disk.device.capacity
+        - disk.device.allocation
+        - disk.device.usage
+    - name: ipmi_pollster
+      interval: 600
+      meters:
+        - hardware.ipmi.node.power
+        - hardware.ipmi.node.temperature
+        - hardware.ipmi.node.outlet_temperature
+        - hardware.ipmi.node.airflow
+        - hardware.ipmi.node.cups
+        - hardware.ipmi.node.cpu_util
+        - hardware.ipmi.node.mem_util
+        - hardware.ipmi.node.io_util
+        - hardware.ipmi.temperature
+        - hardware.ipmi.voltage
+        - hardware.ipmi.current
+        - hardware.ipmi.fan
+    - name: ceph_pollster
+      interval: 600
+      meters:
+        - radosgw.objects
+        - radosgw.objects.size
+        - radosgw.objects.containers
+        - radosgw.api.request
+        - radosgw.containers.objects
+        - radosgw.containers.objects.size
+    - name: image_pollster
+      interval: 600
+      meters:
+        - image.size
+    - name: volume_pollster
+      interval: 600
+      meters:
+        - volume.size
+        - volume.snapshot.size
+        - volume.backup.size
-- 
2.7.4

